File: rust/src/models/checkout_type.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(
    Clone, Copy, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
)]
pub enum CheckoutType {
    #[default]
    #[serde(rename = "SELF_SERVE")]
    SelfServe,

    #[serde(rename = "SUBSCRIPTION_ACTIVATION")]
    SubscriptionActivation,

    #[serde(rename = "PLAN_CHANGE")]
    PlanChange,

    #[serde(rename = "ADDON_PURCHASE")]
    AddonPurchase,
}

impl fmt::Display for CheckoutType {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::SelfServe => "SELF_SERVE",
            Self::SubscriptionActivation => "SUBSCRIPTION_ACTIVATION",
            Self::PlanChange => "PLAN_CHANGE",
            Self::AddonPurchase => "ADDON_PURCHASE",
        };
        f.write_str(value)
    }
}

File: typescript/src/models/checkoutType.ts
// this file is @generated

export enum CheckoutType {
  SelfServe = "SELF_SERVE",
  SubscriptionActivation = "SUBSCRIPTION_ACTIVATION",
  PlanChange = "PLAN_CHANGE",
  AddonPurchase = "ADDON_PURCHASE",
}

export const CheckoutTypeSerializer = {
  _fromJsonObject(object: any): CheckoutType {
    return object;
  },

  _toJsonObject(self: CheckoutType): any {
    return self;
  },
};
